pkg/pillar/cmd/zedrouter/probe.go

```diff
@@ -519,31 +516,33 @@ func probeCheckStatusUseType(status *types.NetworkInstanceStatus, free bool) {
 	}
 
 	if currIntf != "" {
-		// when we have more than two cost levels, this need to change to compare numbers
-		// if the current intf is non-free, and we are in free loop, to see if we can get free UP intf
-		if !status.PInfo[currIntf].IsFree && free {
+		// when we have more than two cost prlevels, this need to change to compare numbers
+		// if the current intf is less desirable than the current loop in the caller, then see if we can get an interface as that priority
+		if status.PInfo[currIntf].FallbackPriority > priority {
 			for _, info := range status.PInfo {
-				if free != info.IsFree {
+				if priority != info.FallbackPriority {
 					continue
 				}
 				if infoUpCount(info) == 0 {
 					continue
 				}
-				log.Tracef("probeCheckStatusUseType: currintf is non-free, randomly assign %s now\n", info.IfName)
+				log.Tracef("probeCheckStatusUseType: currintf was not best priority %d, randomly assign %s now",
+					priority, info.IfName)
 				currIntf = info.IfName
 				break
 			}
 		}
 		currinfo := status.PInfo[currIntf]
 		numOfUps = infoUpCount(currinfo)
 		log.Tracef("probeCheckStatusUseType: curr intf %s, num ups %d\n", currIntf, numOfUps)
-		if free == currinfo.IsFree && numOfUps == 2 { // good, no need to change
+		if priority == currinfo.FallbackPriority && numOfUps == 2 { // good, no need to change
 			status.CurrentUplinkIntf = currIntf
 			return
 		}
 		log.Tracef("probeCheckStatusUseType: before loop\n")
 		for _, info := range status.PInfo {
-			if free != info.IsFree {
+			// XXX should this instead compare for less than?
```

Comment:
i think this '!=' is fine here, since we walk the priority outside this func from low to high number, as long as we can find in this same priority with the highest 'UP' count info, we should use that. Looks good here.